### PR TITLE
feature/musical instrument redesign

### PR DESCRIPTION
**Pull request by ruben-svensson**
### This PR add's a new layout and new sounds to the piano and saxophone as seen by the following examples: 

Full range piano

https://user-images.githubusercontent.com/29916243/134821968-16065725-2986-4407-8e87-3f3d32f15261.mp4

Full range saxophone

https://user-images.githubusercontent.com/29916243/134821980-f0d3f594-1369-445d-ba9a-a2dc94aecef2.mp4

Piano example

https://user-images.githubusercontent.com/29916243/134821988-e7eb2195-59ab-4d72-886c-56fcd768cd1a.mp4

Saxophone example

https://user-images.githubusercontent.com/29916243/134821986-ba1acdef-1883-4980-a6b1-74998f133036.mp4

The piano has been equipped with a total of 39 notes and the saxophone 30, I've gone through every single one of them and checked them (using FL Studio's tool Edison) to make sure they sound good, kept the file size low while not lowering the quality and made sure the volume is the same as the original.

My plan is to get every instrument updated with the new interface and new sound in the future.

The sound has been sourced from the following places:
Piano: https://www.reddit.com/r/piano/comments/3u6ke7/heres_some_midi_and_mp3_files_for_individual/ (by /u/SingleInfinity)
Saxophone: https://freesound.org/people/MTG/packs/20247/ (by MTG)

## Why's this needed? 
This PR will fix what multiple people have stated that the current layout of the notes is difficult to use, for example with the trumpet:
image
The current instruments also lack enough notes to create most songs, which this PR will fix for the piano and saxophone.

## Changelog

```changelog
(u)Grokberg
(+)Adds a new interface and new sound to the piano and saxophone.
```

### CODE REVIEW

**Comment by Gannets** [comment]
This is much better!

File: code/modules/interface/multiContext/context_layouts.dm [hunk 1]
```diff
@@ -125,6 +125,83 @@ var/list/datum/contextAction/globalContextActions = null
...
+		spacingY = SpacingY
+		offsetX = OffsetX
+		offsetY = OffsetY
+		keyOffset = KeyOffset
+		. = ..()
+
+	showButtons(list/buttons, atom/target)
+		var/offX = 0
+		var/offY = spacingY
+		var/finalOff = spacingX * (buttons.len-3)
```
Comment (Sovexe) [inline code/modules/interface/multiContext/context_layouts.dm:146]:
```suggestion
		var/finalOff = spacingX * (length(buttons)-3)
```

File: code/modules/interface/multiContext/context_layouts.dm [hunk 2]
```diff
@@ -125,6 +125,83 @@ var/list/datum/contextAction/globalContextActions = null
...
+					trans.Translate(offX, offY)
+
+					animate(C, alpha=255, transform=trans, easing=CUBIC_EASING, time=1)
+
+			buttonIndex += 1
+
+			offX += spacingX
+			//if(offX >= spacingX)
+			//	offX = 0
+			//	offY -= spacingY
```
Comment (Sovexe) [inline code/modules/interface/multiContext/context_layouts.dm:191]:
```suggestion
```

File: code/modules/interface/multiContext/context_layouts.dm [hunk 3]
```diff
@@ -125,6 +125,83 @@ var/list/datum/contextAction/globalContextActions = null
...
+
+		for(var/i in 1 to blackKeys.len)
```
Comment (Sovexe) [inline code/modules/interface/multiContext/context_layouts.dm:193]:
```suggestion
		for(var/i in 1 to length(blackKeys))
```

File: code/obj/item/instruments.dm [hunk 4]
```diff
@@ -142,16 +154,20 @@
 	icon_state = "piano"
 	item_state = "piano"
 	sounds_instrument = null
-	special_index = 13
+	// special_index = 13
```
Comment (Sovexe) [inline code/obj/item/instruments.dm:157]:
```suggestion
```

File: code/obj/item/instruments.dm [hunk 5]
```diff
@@ -142,16 +154,20 @@
...
 	note_time = 0.18 SECONDS
 	randomized_pitch = 0
+	use_new_interface = 1
 
 	New()
+		notes = list("c4","c-4", "d4", "d-4", "e4","f4","f-4","g4", "g-4","a4","a-4","b4","c5","c-5", "d5", "d-5", "e5","f5","f-5","g5", "g-5","a5","a-5","b5","c6","c-6", "d6", "d-6", "e6","f6","f-6","g6", "g-6","a6","a-6","b6","c7")
 		sounds_instrument = list()
-		for (var/i in 1 to 12)
-			sounds_instrument += "sound/musical_instruments/piano/piano_[i].ogg"
+		for (var/i in 1 to 37) // 12
```
Comment (Sovexe) [inline code/obj/item/instruments.dm:165]: The magic number 37 should be replaced by a define or a length proc

File: code/obj/item/instruments.dm [hunk 6]
```diff
@@ -205,11 +221,16 @@
...
+	use_new_interface = 1
+	//Start at G
+	key_offset = 8
 
 	New()
+		notes = list("g3","g-3","a3","a-3","b3","c4","c-4", "d4", "d-4", "e4","f4","f-4","g4", "g-4","a4","a-4","b4","c5","c-5", "d5", "d-5", "e5","f5","f-5","g5", "g-5","a5","a-5","b5","c6")
 		sounds_instrument = list()
-		for (var/i in 1 to 12)
-			sounds_instrument += "sound/musical_instruments/sax/sax_[i].ogg"
+		for (var/i in 1 to 29) // 12
```
Comment (Sovexe) [inline code/obj/item/instruments.dm:231]: is 29 correct here, there are 30 notes in the above list. 

This magic number 29 should be replaced by a define or a length proc

File: code/modules/interface/multiContext/context_layouts.dm [hunk 7]
```diff
@@ -125,6 +125,83 @@ var/list/datum/contextAction/globalContextActions = null
...
+		. = ..()
+
+	showButtons(list/buttons, atom/target)
+		var/offX = 0
+		var/offY = spacingY
+		var/finalOff = spacingX * (buttons.len-3)
+		offX -= finalOff/2
+
+		var/buttonIndex = keyOffset
+		// var/octaveIndex = 1
```
Comment (Sovexe) [inline code/modules/interface/multiContext/context_layouts.dm:150]:
```suggestion
```

**Review by Sovexe — CHANGES_REQUESTED** [review]

**Review by virvatuli-gh — APPROVED** [review]
Not commenting on the code aspect of this but it's a great idea and all the sound-related stuff seems A-OK to me! Letting people have a wider range of notes to make some cool tunes is good in my books.

**Comment by ruben-svensson** [comment]
Thanks for suggestions @Sovexe , please check again if I did it correct. :)

File: code/modules/interface/multiContext/context_layouts.dm (hunk 1 above)
Comment (ruben-svensson) [reply code/modules/interface/multiContext/context_layouts.dm]: Change made

**Review by Sovexe — APPROVED** [review]

**Pull request merged**